studio/data/edge-functions/edge-functions-update-mutation.tsx

```diff
@@ -0,0 +1,53 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { patch } from 'lib/common/fetch'
+import { API_ADMIN_URL } from 'lib/constants'
+import { edgeFunctionsKeys } from './keys'
+
+export type EdgeFunctionsUpdateVariables = {
+  projectRef: string
+  slug: string
+  payload: {
+    name?: string
+    verifyJwt?: boolean
+    importMap?: boolean
+  }
+}
+
+export async function updateEdgeFunction({
+  projectRef,
+  slug,
+  payload,
+}: EdgeFunctionsUpdateVariables) {
+  if (!projectRef) throw new Error('projectRef is required')
+
+  const response = await patch(`${API_ADMIN_URL}/projects/${projectRef}/functions/${slug}`, payload)
+  if (response.error) throw response.error
+
+  return response
+}
+
+type EdgeFunctionsUpdateData = Awaited<ReturnType<typeof updateEdgeFunction>>
+
+export const useEdgeFunctionUpdateMutation = ({
+  onSuccess,
+  ...options
+}: Omit<
+  UseMutationOptions<EdgeFunctionsUpdateData, unknown, EdgeFunctionsUpdateVariables>,
+  'mutationFn'
+> = {}) => {
+  const queryClient = useQueryClient()
+
+  return useMutation<EdgeFunctionsUpdateData, unknown, EdgeFunctionsUpdateVariables>(
+    (vars) => updateEdgeFunction(vars),
+    {
+      async onSuccess(data, variables, context) {
+        const { projectRef } = variables
+        await queryClient.invalidateQueries(edgeFunctionsKeys.list(projectRef), {
+          refetchType: 'all',
```

Comment:
Any reason you're choosing to do a `refetchType: 'all'` here? We're not redirecting back to the list page or anything like that here are we?

Reply:
oh right we're not, was running the assumption that we needed to toggle a refetch to fetch the list of functions again so that its updated. i'm guessing that the queryClient for functions will automatically refetch when we go back to the functions list page?

Reply:
Yep exactly. It's still a good idea to mark it as invalidated with `invalidateQueries()` in case we change any caching rules in future.